### PR TITLE
3. GET /api/categories

### PR DESCRIPTION
**Pull request by dafyddhenke**
Relevant card: https://trello.com/c/76pXkD3u

Completed 3. GET /api/categories task with:

- listen file for locally testing server
- app file for controlling express server endpoints
- model and controller for getting categories
- JEST tests for testing app (e.g. rejects invalid endpoints)
- JEST tests for getting categories)

### CODE REVIEW

File: __tests__/app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const request = require("supertest");
```
Comment (two-triangles) [inline __tests__/app.js:1]: please keep your test files in the format of <name>.test.js 

 - you have two files called app.js and this is a bit misleading. 
    - app.test.js would be more suitable here

File: __tests__/categories.controller.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      .expect(200)
+      .then((response) => {
+        expect(Array.isArray(response.body)).toBe(true);
+      });
+  });
+  test("GET /api/categories should return an an array of objects with slug and description props", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then((response) => {
```
Comment (two-triangles) [inline __tests__/categories.controller.js:31]: we should be asserting how many categories come back before checking them

File: __tests__/app.js (hunk 1 above)
Comment (dafyddhenke) [reply __tests__/app.js:1]: Fixed in next request.

**Comment by two-triangles** [comment]
Works looks good here

  - please change the names of test files to the correct format 
  - add a check for the correct amount of topics coming back 

When you commit this it will update this pr and I am happy for you to merge at that point

**Comment by dafyddhenke** [comment]
> add a check for the correct **amount of topics** coming back

Unless told otherwise, I'll assume you mean categories.

File: __tests__/categories.controller.js (hunk 2 above)
Comment (dafyddhenke) [reply __tests__/categories.controller.js:31]: Investigating now.

**Pull request merged**